boa_engine/src/vm/opcode/mod.rs

```diff
@@ -1,4 +1,77 @@
 /// The opcodes of the vm.
+use crate::{vm::ShouldExit, Context, JsResult};
+
+// Operation modules
+pub(crate) mod await_stm;
+pub(crate) mod binary_ops;
+pub(crate) mod call;
+pub(crate) mod concat;
+pub(crate) mod copy;
+pub(crate) mod define;
+pub(crate) mod delete;
+pub(crate) mod dup;
+pub(crate) mod environment;
+pub(crate) mod generator;
+pub(crate) mod get;
+pub(crate) mod iteration;
+pub(crate) mod jump;
+pub(crate) mod new;
+pub(crate) mod nop;
+pub(crate) mod pop;
+pub(crate) mod promise;
+pub(crate) mod push;
+pub(crate) mod require;
+pub(crate) mod rest_parameter;
+pub(crate) mod return_stm;
+pub(crate) mod set;
+pub(crate) mod swap;
+pub(crate) mod switch;
+pub(crate) mod throw;
+pub(crate) mod to;
+pub(crate) mod try_catch;
+pub(crate) mod unary_ops;
+pub(crate) mod value;
+pub(crate) mod void;
+
+// Operation structs
+pub(crate) use await_stm::*;
+pub(crate) use binary_ops::*;
+pub(crate) use call::*;
+pub(crate) use concat::*;
+pub(crate) use copy::*;
+pub(crate) use define::*;
+pub(crate) use delete::*;
+pub(crate) use dup::*;
+pub(crate) use environment::*;
+pub(crate) use generator::*;
+pub(crate) use get::*;
+pub(crate) use iteration::*;
+pub(crate) use jump::*;
+pub(crate) use new::*;
+pub(crate) use nop::*;
+pub(crate) use pop::*;
+pub(crate) use promise::*;
+pub(crate) use push::*;
+pub(crate) use require::*;
+pub(crate) use rest_parameter::*;
+pub(crate) use return_stm::*;
+pub(crate) use set::*;
+pub(crate) use swap::*;
+pub(crate) use switch::*;
+pub(crate) use throw::*;
+pub(crate) use to::*;
+pub(crate) use try_catch::*;
+pub(crate) use unary_ops::*;
+pub(crate) use value::*;
+pub(crate) use void::*;
+
+pub(crate) trait Operation {
+    const NAME: &'static str;
+    const INSTRUCTION: &'static str;
+
+    fn execute(context: &mut Context) -> JsResult<ShouldExit>;
+}
```

Comment:
This trait should have some documentation, to clearly explain what does it do, how would someone implement it for a new opcode and so on.

Ideally, all implementing opcodes should have module-level and structure-level documentation, but we are lacking documentation in general for the VM (look at the `Opcode` enum just below). If you feel that documenting all op codes is a bit too much for this PR, let me know, and I'll open a new issue (similar to #1608) to document the VM.